home/mixins.py

```diff
@@ -61,10 +62,25 @@ def _get_stream_data_image_urls(self, raw_data):
 
         return image_urls
 
+    def _get_stream_data_media_urls(self, stream_data):
+        media_urls = []
+
+        for block in stream_data:
+            if block['type'] == 'media':
+                media = Media.objects.filter(id=block['value']).first()
+                if media:
+                    media_urls.append(media.url)
+
+        return media_urls
+
     @property
     def get_image_urls(self):
         raise NotImplementedError
 
+    @property
+    def get_media_urls(self):
+        raise NotImplementedError
```

Comment:
Why raise an exception by default and not, say, an empty list?

Why treat images and media separately instead of having a method to collect all URLs?